task-driver/src/settle_match_internal.rs

```diff
@@ -385,9 +385,19 @@ impl SettleMatchInternalTask {
         }
         let tx_hash =
             tx_submit_res.map_err(|err| SettleMatchInternalTaskError::Starknet(err.to_string()))?;
-        log::info!("tx hash: 0x{tx_hash:x}");
 
-        // TODO: Fix the polling method and await finality here
+        log::info!("tx hash: 0x{tx_hash:x}");
+        let status = self
+            .starknet_client
+            .poll_transaction_completed(tx_hash)
+            .await
+            .map_err(|err| SettleMatchInternalTaskError::Starknet(err.to_string()))?;
+        if status == TransactionStatus::Reverted || status == TransactionStatus::Aborted {
```

Comment:
could we use `status.into_result().map_err(...)` here?

Reply:
No, we do not want to error on `Rejected` in the match tasks. The transaction rejects for one of the parties if both try to settle, this case is handled above.